src/main/scala/mrpowers/jodie/DeltaHelpers.scala

```diff
@@ -339,6 +339,65 @@ object DeltaHelpers {
       .save(storagePath)
   }
 
+  /**
+   * Validates and appends data to a Delta table. This function ensures that the provided DataFrame can be appended
+   * to the specified Delta table by checking data type compatibility and column presence.
+   *
+   * @param deltaTable   The Delta table to which data will be appended.
+   * @param appendDF     The DataFrame containing data to be appended.
+   * @param requiredCols The list of required columns in the appendDF.
+   * @param optionalCols The list of optional columns in the appendDF.
+   * @throws IllegalArgumentException if input arguments have an invalid type, are missing, or are empty.
+   * @throws IllegalArgumentException if required columns are missing in the provided Delta table.
+   * @throws IllegalArgumentException if a column in the append DataFrame is not part of the original Delta table.
+   */
+  def validateAppend(
+                      deltaTable: DeltaTable,
+                      appendDF: DataFrame,
+                      requiredCols: List[String],
+                      optionalCols: List[String]
+                    ): Unit = {
+    // Check if deltaTable is an instance of DeltaTable
+    if (!deltaTable.isInstanceOf[DeltaTable]) {
```

Comment:
Because scala is a typed language you don't need to add this type of checking, if you try to use this function with a different type it will fail to compile.